### ISSUE TITLE
"File not found" error when package and directory structure do not match

### ISSUE THREAD

**Issue by nclinger** [protostuff/protobuf-jetbrains-plugin#87]
There is no clear reason why the package and directory structures have to match, and is a non-requirements as far as the proto team are concerned (from what I know).  So why is the plugin showing an error for this?

Thanks!

**Comment by kshchepanovskyi** [comment]
It's not a requirement, but that way plugin is able to resolve imports without any additional configuration.
Plugin tries to resolve import against one of "source roots", and if it fails to, it shows an error.
Next plugin release will contain better heuristics for import resolution.

> The protocol compiler searches for imported files in a set of directories specified on the protocol compiler command line using the -I/--proto_path flag. If no flag was given, it looks in the directory in which the compiler was invoked. In general you should set the --proto_path flag to the root of your project and use fully qualified names for all imports.

https://developers.google.com/protocol-buffers/docs/proto3#other

`protoc` resolves imports against "include paths" which are explicitly specified. For a plugin, it is a little bit more complicated, because it should "guess" include paths (except if they are specified in "custom include paths" settings).

All lookups are registered here: https://github.com/protostuff/protobuf-jetbrains-plugin/blob/master/src/main/java/io/protostuff/jetbrains/plugin/reference/file/FilePathReferenceProvider.java#L64

Last change in resolution algorithm that I did - I added https://github.com/protostuff/protobuf-jetbrains-plugin/blob/master/src/main/java/io/protostuff/jetbrains/plugin/reference/file/ProtoFileRelativePathRootsProvider.java - it uses package statement in order to guess "include path". I hoped that it will cover most of the cases, but apparently it does not, so it is still "work in progress".
